components/script/dom/readablestream.rs

```diff
@@ -627,6 +694,129 @@ impl ReadableStream {
             },
         }
     }
+    /// <https://streams.spec.whatwg.org/#abstract-opdef-readablestreamdefaulttee>
+    #[allow(crown::unrooted_must_root)]
+    fn default_tee(
+        &self,
+        clone_for_branch_2: bool,
+        can_gc: CanGc,
+    ) -> Fallible<Vec<DomRoot<ReadableStream>>> {
+        // Assert: stream implements ReadableStream.
+
+        // Assert: cloneForBranch2 is a boolean.
+        let clone_for_branch_2 = Rc::new(Cell::new(clone_for_branch_2));
+
+        // Let reader be ? AcquireReadableStreamDefaultReader(stream).
+        let reader = self.acquire_default_reader(can_gc)?;
+        self.set_reader(Some(&reader));
+
+        // Let reading be false.
+        let reading = Rc::new(Cell::new(false));
+        // Let readAgain be false.
+        let read_again = Rc::new(Cell::new(false));
+        // Let canceled1 be false.
+        let canceled_1 = Rc::new(Cell::new(false));
+        // Let canceled2 be false.
+        let canceled_2 = Rc::new(Cell::new(false));
+
+        let cx = GlobalScope::get_cx();
+        // Let reason1 be undefined.
+        rooted!(in(*cx) let mut reason_1 = UndefinedValue());
+        // Let reason2 be undefined.
+        rooted!(in(*cx) let mut reason_2 = UndefinedValue());
+        // Let branch1 be undefined.
+        let branch_1 = MutNullableDom::new(None);
+        // Let branch2 be undefined.
+        let branch_2 = MutNullableDom::new(None);
+        // Let cancelPromise be a new promise.
+        let cancel_promise = Promise::new(&self.reflector_.global(), can_gc);
+
+        let underlying_source_type_branch_1 = UnderlyingSourceType::Tee(TeeUnderlyingSource::new(
```

Comment:
I think we should also reflect the `TeeUnderlyingSource`, and make `UnderlyingSourceType::Tee` take a `Dom`.